tuning/utils/preprocessing_utils.py

```diff
@@ -29,8 +29,48 @@
 logger = logging.get_logger("sft_trainer_preprocessing")
 
 
+def _is_pretokenized_dataset(dataset: Dataset):
```

Comment:
i suggest move this to inside `validate_data_args` and put a comment that his is how SFT checks for tokenized internally.

Reply:
@fabianlim done